check_diff/src/lib.rs

```diff
@@ -1,11 +1,63 @@
 use std::env;
 use std::io;
 use std::path::Path;
+use std::path::PathBuf;
 use std::process::Command;
+use std::process::Output;
 use tracing::info;
 
+pub enum CheckDiffError {
+    FailedGit(GitError),
+    FailedCommand(String),
+    FailedUtf8(String),
```

Comment:
Instead of storing a string here, why not just store the [Utf8Error](https://doc.rust-lang.org/std/str/struct.Utf8Error.html#) we get if the conversion fails?

Should be another simple `From` conversion that we can implement to go from `Utf8Error` -> `CheckDiffError::FailedUtf8` similar to how we're already doing it for `IO` Errors.